vignettes/how-to-update-released-site.Rmd

```diff
@@ -0,0 +1,272 @@
+---
+title: "How to update a released site"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{How to update a released site}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Checklist
+
+If you're already up-to-speed, here's the checklist:
+
+* Update to usethis >= v2.1.5.9000, maybe with
+  `pak::pak("r-lib/usethis")`.
+* Update pkgdown to its latest version, maybe with `pak::pak("pkgdown")`.
+* Check out your default branch and `git pull`.
+* Call `usethis::use_pkgdown_github_pages()`. You may want to do this even if
+  you are already using pkgdown + GitHub Pages + GitHub Actions. However, you
+  will need to inspect the changes and use your judgement when staging /
+  discarding individual diffs. Intended benefits:
+  - Update `.github/workflows/pkgdown.yaml` to the latest version from
+    `r-lib/actions`. (This specific step can also be achieved with
+    `usethis::use_github_action("pkgdown")` or
+    `usethis::use_tidy_github_actions()`.)
+  - Make sure the pkgdown site's URL appears in `_pkgdown.yml`, in the URL
+    field of DESCRIPTION, and as the website associated with the GitHub repo.
+  - Add "tidyverse/tidytemplate" to `Config/Needs/website` in DESCRIPTION, if
+    repo is owned by one of these orgs: tidyverse, r-lib, tidymodels, rstudio. 
+* Make sure you're using automatic development mode, i.e. in `_pkgdown.yml`:
+
+  ``` yaml
+  development:
+    mode: auto
+  ```
+* Modernize `_pkgdown.yml` following the instructions here:
+  - <https://tidytemplate.tidyverse.org>
+* Commit and push, if any of the above caused local changes.
+* Scrutinize your newly published dev site and make sure things look right.
+* Create and checkout a branch based on the tag of your most recent GitHub/CRAN
+  release.
+  
+  Example for readr, whose latest release is 2.1.1:  
+  `git checkout -b update-pkgdown-2-1-1 v2.1.1`
+  
+  General pattern:  
+  `git checkout -b NEW-BRANCH-NAME NAME-OF-RELEASE-TAG`
+* Backport innovations from the future. Most likely candidates:
+  - `_pkgdown.yml`
+  - `.github/workflows/pkgdown.yaml`
+  - `Config/Needs/website` field of DESCRIPTION
+  - `README.Rmd` / `README.md`. Do you need `devtools::build_readme()`?
+  - Any fixes to released roxygen comments / docs, then `document()`.
+  - Any new vignettes or articles that apply to the released version.
+  
+  Useful git commands
+  ```
+  git cherry-pick SHA
+  git checkout REF -- path/to/a/specific/file
+  ```
+* Push this branch, maybe using `usethis::pr_push()`.
+* Go to your package's GHA page, maybe using `usethis::browse_github_actions()`.
+  Select the pkgdown workflow. Click *Run workflow* and select the branch you
+  just pushed.
+* Scrutinize your newly published **released** site and make sure things look
+  right.
```

Comment:
Then what do you do with the branch/PR? Just delete it?

Reply:
Again, I say this below in the longer form.

> You can keep this branch around for a while, in case you didn't get everything right the first time or if more things crop up that you'd like backport to the released site, before your next CRAN release.

Should I copy that in the TL;DR?